front/app/components/EventAttendanceButton/index.tsx

```diff
@@ -140,87 +134,65 @@ const EventAttendanceButton = ({ event }: EventAttendanceButtonProps) => {
     setConfirmationModalVisible(true);
   };
 
```

Comment:
The main change is splitting this component into two buttons: one for when you're attending or registered for an event, and another for when you're not. This eliminates some of the bloated conditionals.